### PR TITLE
Assorted doc improvements

### PR DESCRIPTION
**Pull request by bjoernQ**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
This improves some API documentation - mostly for RMT and some others

#### Testing
Build the docs and have a look

### CODE REVIEW

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks for this sweet documentation improvements!

File: esp-hal/src/lib.rs [hunk 1]
```diff
@@ -70,6 +70,49 @@
 // MUST be the first module
 mod fmt;
 
+#[cfg(esp32)]
```
Comment (MabezDev) [inline esp-hal/src/lib.rs:73]: Should we move this to the soc module for each chip?

File: esp-hal/src/rmt.rs [hunk 2]
```diff
@@ -634,6 +645,7 @@ mod impl_for_chip {
     use crate::peripheral::{Peripheral, PeripheralRef};
 
     /// RMT Instance
+    #[allow(missing_docs)]
```
Comment (MabezDev) [inline esp-hal/src/rmt.rs:648]: Why is this needed? For the inner `pub` channels? Maybe we should just doc them?

**Review by MabezDev — CHANGES_REQUESTED** [review]
Thank you!! This is an amazing improvement! I have a couple of questions though.

File: esp-hal/src/rmt.rs (hunk 2 above)
Comment (bjoernQ) [reply esp-hal/src/rmt.rs:648]: Yes, it's for the channels inside the struct. 

My thinking is that having "Channel 0","Channel 1" etc. as a comment on every channel is somewhat boring and not very useful 🤔

File: esp-hal/src/lib.rs (hunk 1 above)
Comment (bjoernQ) [reply esp-hal/src/lib.rs:73]: That's a good idea! 💡
